### PR TITLE
feat: use GetLogEntryByIndex to query rekor

### LINKED ISSUE #52: Use GetLogEntryByIndex when querying Rekor
**Description**

Relevant code: https://github.com/sigstore/sigstore-go/blob/9a59c7f51a05d2305a8ba4ab9343416aaaeb16d6/pkg/verify/tlog.go#L135-L138

SearchLogQuery is used when querying Rekor to get an entry, which returns a list of entries matching the provided index. This will return a single entry since we are requesting a single index, but we can instead use GetLogEntryByIndex which will always return a single log entry.

(This is a low priority improvement, not a functional one)

### PR DESCRIPTION
**Pull request by vishal-chdhry**
#### Summary

Closes: #52

Replaces SearchLogQuery with GetLogEntryByIndex which always returns a single log entry.

#### TODO
Add unit tests
#### Release Note
NONE

#### Documentation
NONE

### CODE REVIEW

**Comment by vishal-chdhry** [comment]
@steiza I could not find any unit test for online set to `true`, any tips on how should I test it?

**Review by steiza — DISMISSED** [review]
Yeah, testing could definitely be easier. Until we fix #53, I think the best way to test this is to get a bundle and run a verification manually.

There's a bundle in `examples/bundle-provenance.json`. You can also get bundles from many npm packages with something like:

```
$ curl https://registry.npmjs.org/-/npm/v1/attestations/semver@7.6.1 | jq ".attestations[0]"
```

Once you have the bundle, you can inspect the certificate to find the SAN and the issuer (`https://token.actions.githubusercontent.com` for GitHub Actions), and then attempt to verify the bundle with online tlog lookup via:

```
sigstore-go$ go run cmd/sigstore-go/main.go -expectedSAN "https://github.com/sigstore/sigstore-js/.github/workflows/release.yml@refs/heads/main" -expectedIssuer "https://token.actions.githubusercontent.com" -onlineTlog examples/bundle-provenance.json
...
Verification successful!
```

LGTM!

**Comment by vishal-chdhry** [comment]
Okay, I will look into #53 and also add a test for this with that

**Review by Hayden-IO — DISMISSED** [review]
Thanks for the change!

**Comment by Hayden-IO** [comment]
Before I merge, to confirm, was this tested locally?

**Comment by vishal-chdhry** [comment]
tested it out with the example provided by @steiza and it passed
```
go run cmd/sigstore-go/main.go -expectedSAN "https://github.com/sigstore/sigstore-js/.github/workflows/release.yml@refs/heads/main" -expectedIssuer "https://token.actions.githubusercontent.com" -onlineTlog examples/bundle-provenance.json
```

But I think we should add tests for this. Since this is not functional, we can pause it, I will look into #53 in the weekend and add tests for this too

**Comment by Hayden-IO** [comment]
Thank you!

File: pkg/verify/tlog.go [hunk 1]
```diff
@@ -148,7 +144,7 @@ func VerifyArtifactTransparencyLog(entity SignedEntity, trustedMaterial root.Tru
 				return nil, errors.New("too many log entries returned")
```
Comment (Hayden-IO) [inline pkg/verify/tlog.go]: @vishal-chdhry can you remove this error condition, as it's no longer relevant?

**Comment by vishal-chdhry** [comment]
@haydentherapper done

**Review by Hayden-IO — APPROVED** [review]

**Pull request merged**